File: Python/ByRound/2059/2059_E1_Stop_Gaming_Easy_Version.py
# Problem: CF 2059 E1 - Stop Gaming (Easy Version)
# https://codeforces.com/contest/2059/problem/E1

"""
E1. Stop Gaming (Easy Version)

Algorithm:
- This problem involves transforming arrays using operations that insert elements at the beginning of arrays.
- The key insight is to use a greedy approach: for each element in the target array,
  determine whether it matches the current position in the original array, and if not,
  count how many elements would be "wasted" by shifting.
- We process elements in order from top to bottom, left to right. For each row:
    - Try to match elements with the original array.
    - If there's a mismatch, we must perform an operation on a previously processed array.
    - The number of operations is computed as: total elements - matched elements.

Time Complexity: O(n * m)
Space Complexity: O(1) - only using a few variables for tracking state.

"""
import sys

input = lambda: sys.stdin.readline().rstrip()


def solve_case():
    n, m = map(int, input().split())
    a = [list(map(int, input().split())) for _ in range(n)]
    b = [list(map(int, input().split())) for _ in range(n)]
    used = 0  # Count of matched elements
    ra, ca = 0, 0  # Row and column index in original array
    for rb in range(n):  # Iterate through rows of target array
        for cb in range(m):  # Iterate through columns of target array
            if b[rb][cb] == a[ra][ca]:  # If element matches current position
                used += 1  # Increment matched count
                ca += 1  # Move to next column in original array
                if ca == m:  # If we've reached end of row
                    ra += 1  # Move to next row
                    ca = 0  # Reset column index
            elif ra == rb:  # If mismatch and we're at same row as original
                used -= ca  # Subtract the number of elements we can't use from current row
                ca = 0  # Reset column index
    ans = n * m - used  # Total elements minus matched elements is the operations count
    return ans


ans = []
for _ in range(int(input())):
    ans += [str(solve_case())]
print("\n".join(ans))


# https://github.com/VaHiX/codeForces/

File: CPP/ByTier/E/2059_E1_Stop_Gaming_Easy_Version.cpp
// Problem: CF 2059 E1 - Stop Gaming (Easy Version)
// https://codeforces.com/contest/2059/problem/E1

/*
E1. Stop Gaming (Easy Version)
Algorithms/Techniques: Simulation, Two Pointers

Time Complexity: O(n * m) amortized per test case
Space Complexity: O(n * m)

The problem involves transforming n arrays of size m each from an initial state 'a' to a final state 'b'
using a specific operation where we can prepend an element to arrays and shift following arrays.

This solution uses a simulation approach with two pointers to find the minimum number of operations
by comparing elements in both arrays and tracking mismatches.
*/
#include <iostream>
using namespace std;
const int N = 3e5 + 10;
int a[N], b[N];

void solve() {
  int n, m;
  cin >> n >> m;
  for (int i = 0; i < n * m; i++)
    cin >> a[i];
  for (int i = 0; i < n * m; i++)
    cin >> b[i];
  
  int pa = 0, pb = 0, flag = 1, ans = n * m;
  
  while (flag) {
    // Move pb forward until we find a matching element or reach end
    while (pb != n * m && b[pb] != a[pa])
      pb++;
    
    if (pb == n * m)
      break; // No more matches, stop searching
    
    // If elements are in the same row (array)
    if (pa / m == pb / m) {
      pa++, pb++; // Advance both pointers
      
      // Check if current row elements match
      while (pa % m != 0 && pa / m == pb / m) {
        if (a[pa] != b[pb]) {
          flag = 0; // Mismatch found, stop
          break;
        }
        pa++, pb++; // Move to next element in row
      }
    } else {
      // If elements are in different rows, only advance both pointers
      pa++, pb++;
    }
    
    if (flag)
      ans = n * m - pa; // Update answer with remaining elements to process
  }
  
  cout << ans << endl;
  return;
}

int main() {
  ios_base::sync_with_stdio(false);
  cin.tie(0), cout.tie(0);
  int _;
  cin >> _;
  while (_--)
    solve();
  return 0;
}


// https://github.com/VaHiX/codeForces/